src/hooks/useCart/useCart.tsx

```diff
@@ -323,8 +324,19 @@ export const useCart = (
         setCheckoutResult(transactionResponse);
         setCartState("PURCHASED");
       } catch (e) {
+        if (e.message === ERROR_MESSAGE.DUPLICATE_POD_INPUT) {
+          setError(e);
+        } else {
+          setError(new Error(ERROR_MESSAGE.SERVER_ERROR));
+        }
         setCartState("DEFAULT");
-        setError(new Error("Couldn't checkout, please try again later"));
+        if (
+          e.message === "Invalid Purchase Request: Duplicate identifier inputs"
+        ) {
+          setError(new Error(ERROR_MESSAGE.DUPLICATE_IDENTIFIER_INPUT));
+        } else {
+          setError(new Error(ERROR_MESSAGE.SERVER_ERROR));
+        }
```

Comment:
Hmm, would this be better?

```
setCartState("DEFAULT");
if (e.message === ERROR_MESSAGE.DUPLICATE_POD_INPUT) {
  setError(e);
}
else if (
  e.message === "Invalid Purchase Request: Duplicate identifier inputs"
) {
  setError(new Error(ERROR_MESSAGE.DUPLICATE_IDENTIFIER_INPUT));
} 
else {
  setError(new Error(ERROR_MESSAGE.SERVER_ERROR));
}
```

Reply:
ps and thanks!! this one looks a bit funny probably cos of the merge conflicts, fixed!!